### ISSUE TITLE
Is the PyGithub project dead? How can the community help?

### ISSUE THREAD

**Issue by EnricoMi** [PyGithub/PyGithub#2178]
Over the last months, I have seen many very good pull requests from the community that fix blocking bugs or add new features, as well as many questions around issues using this library. This project is actively being used by [25k (public) Github repositories](https://github.com/PyGithub/PyGithub/network/dependents?package_id=UGFja2FnZS01MjQ4NDg2OQ%3D%3D).

Unfortunately, this projects seems to be under-equipped with active maintainers. While the [MAINTAINERS](https://github.com/PyGithub/PyGithub/blob/master/MAINTAINERS) file lists six maintainers, there is only one lonely soul left driving this project. Blocking issues are mounting, ready pull requests are not reviewed, approved or merged, users getting annoyed by the lack of response while Github API is actively developing away from this Python implementation.

This is not sustainable.

Everyone knows and appreciates that this is open-source software and maintainers are putting their free time into this project. But this is no excuse to not improve the situation.

## How can the community help?

💡 I want to collect things that the community can do to revive this project. Please leave your ideas of supporting this project below.

### Things the community can do:
- try hard to minimize the time effort for maintainers:
  - try to investigate the issues yourself, create bug fixes and pull requests
  - try to do the same for other users that don't or can't do it
  - review others' pull requests so that they are more likely to get approved
  - answer other users' question
  - *ideally, maintainers only have to approve and merge pull requests,
    as well as cut releases or give broad guidance on design decisions*
- individuals backed by companies that use this Python package should consider to invest
  some of their work time in contributing back to this open-source software project
- experienced users can help new users getting started
- support maintainer or contributors financially

### Things maintainers can do:
- make the base of maintainers broader
- delegate code reviews to community members that have contributed or code reviewed before
  - invite them directly to specific issues or pull requests
- get completed pull requests merged and released
- focus on on-boarding new maintainers and code reviewers rather than fixing bugs

## Who volunteers?

Please leave a comment below if you volunteer for any of these roles:
- maintainer
- code reviewer
  - get pull requests into a good shape so that maintainers have to spend less time and are more likely to approve
- experienced user
  - answer questions and resolve issues related to usage of the library, not bugs or new features

**Comment by theCapypara** [comment]
Hi! I would also be ready to help, I'm not super familiar with this project yet, aside from some minor work for adding autolink references (#2016), but I would still like to help out however I can.

Can any of the active maintainers say something about the future of this project? Does Github itself have maybe in interest in helping maintaining this project?

**Comment by EnricoMi** [comment]
This project is so actively been used, so many questions, issues and open pull requests. But no attempt is done by the maintainers to open it up to the community.

[MAINTAINERS](https://github.com/PyGithub/PyGithub/blob/master/MAINTAINERS), please mark this project as dead so the community is not wasting more time here.

Community, feel free to fork this repository, get all the good work merged and released there. Post your maintained fork here: https://github.com/PyGithub/PyGithub/discussions/2205

**Comment by sfdye** [comment]
Copying my answer from #2320

> This is fair. Simply put, the maintainers are busy with their lives. If any of you want to get this project moving, consider 
> joining the team or sponsoring via Github sponsors.
>
> However I can get the PR review going when I have some free time towards the end of the year. In particular which ones would like to have first and released to PyPI? Let me know so I can prioritize.

**Comment by EnricoMi** [comment]
@sfdye 

high priority:
#2145 fixing #1989 (#1235, #2113, #2005, #1319, #1115, #810)
#2159 fixing #1399 and #2158

low priority but should also less controversial:
#2147, #2151

**Comment by EnricoMi** [comment]
Alternative to Discussion #2321 is to look at the emoji reaction of pull requests.

GitHub UI allows to sort for :+1:s:
open :+1: https://github.com/PyGithub/PyGithub/pulls?q=is%3Apr+is%3Aopen+sort%3Areactions-%2B1-desc
closed stale :+1: https://github.com/PyGithub/PyGithub/pulls?q=is%3Apr+sort%3Areactions-%2B1-desc+is%3Aclosed+label%3Astale

Some people love to use :heart::
open :heart: https://github.com/PyGithub/PyGithub/pulls?q=is%3Apr+sort%3Areactions-heart-desc+is%3Aopen+
closed stale :heart: https://github.com/PyGithub/PyGithub/pulls?q=is%3Apr+sort%3Areactions-heart-desc+is%3Aclosed+label%3Astale

The community should go to their most wanted pull requests and use the emoji button on the pull request description to add a :+1::
![grafik](https://user-images.githubusercontent.com/44700269/193209089-69daccec-0089-4bb5-a44a-5fa52ecf8fc0.png)
![grafik](https://user-images.githubusercontent.com/44700269/193209893-c322e27e-76df-43fd-874c-b00234afeec4.png)

**Comment by sfdye** [comment]
I will be on vacation from today, but before I go, let me kick off the cadence by fixing the failing CI in #2330. This should unblock the review of many PR because previously the CI failures are almost all related to the dependency issues which I have fixed now. Once you update your branch from master, your build should hopefully be green.

**Comment by sfdye** [comment]
Also I just released [v1.56](https://pypi.org/project/PyGithub/#history)

**Comment by trim21** [comment]
I'd like to become a maintainer to help review and merge pr

**Comment by EnricoMi** [comment]
@mcauley-penney please watch this project (at least issues and discussions) and pick questions as you like:
![image](https://github.com/PyGithub/PyGithub/assets/44700269/0b53151a-f199-4c4b-83f1-910f5133d714)

See you around!

**Comment by lucasgonze** [comment]
I think the answer is that yes, it is dead. It is so dead that the founders can't be bothered to archive it or pass control to new maintainers.

I'm putting this in a comment to warn others who may be trying to figure this out. Woe is me. Do not do as I have done - scaffold a bunch of PyGithub code and then find out that it is way behind the actual API.

**Comment by tomtastico** [comment]
@lucasgonze I have seen @EnricoMi has released a new version 3 weeks ago. So not dead?

But @EnricoMi is the claim that PyGithub is way behind the actual github API warranted? I was planning to start using PyGithub in a new project.

**Comment by EnricoMi** [comment]
PyGithub is and always will be behind the Github API as (maybe countless) Github developers move the API forward and PyGithub follows. The speed to catch up is determined by
1) the community adopting features
2) maintainers reviewing, merging and releasing

Right now, PyGithub has only one active maintainer, so current pace could be improved by
1) having the community do more reviews to reduce effort for maintainers
2) volunteer as a maintainer to govern code quality, shape design and improve overall architecture

Dedicating one hour per day would greatly move this project forward, even one hour per week would be a boost.

**Comment by devstein** [comment]
[Dosu](https://dosu.dev/) could help with community support on issues and discussion

**Comment by MarkForesta** [comment]
@EnricoMi I can review a few PRs if it will help. I have used the library quite extensively at work and feel comfortable being a first look for any of the easy ones.

**Comment by EnricoMi** [comment]
@MarkForesta that would certainly help! A good start would be to review
- pull requests for top issues: https://github.com/PyGithub/PyGithub/issues/3050
- top pull requests: https://github.com/PyGithub/PyGithub/issues/3050
- [recently updated pull requests](https://github.com/PyGithub/PyGithub/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc)
